web/ui/src/common/resourcesFactory.js

```diff
@@ -269,16 +287,56 @@
 
             return function(/* args */){
                 var url = config.url.apply(null, arguments),
-                    payload;
+                    resourceName = url,
+                    payload,
+                    // deferred that will be returned to the user
+                    deferred = $q.defer();
+
+                // if resourceName has query params, strip em off
+                if(resourceName.indexOf("?")){
+                    resourceName = resourceName.split("?")[0];
+                }
+
+                // NOTE: all of our code expects angular's wrapped
+                // promises which provide a success and error method
+                // TODO - remove the need for this
+                httpify(deferred);
+
+                if(pendingGETRequests[resourceName]){
+                    // theres already a pending request to
+                    // this endpoint, so fail!
+                    deferred.reject(`a request to ${resourceName} is pending`);
```

Comment:
Also, does the request-pending message from this reject bubble back up to the user?  Cuz the URL is not likely to be anything that makes sense to the user.  However, at a minimum we should log to the console I think